craft_parts/infos.py

```diff
@@ -131,18 +143,55 @@ def project_options(self) -> Dict[str, Any]:
             "target_arch": self.target_arch,
         }
 
-    def set_custom_argument(self, name: str, value: str) -> None:
-        """Set the value of a custom argument.
+    def set_project_var(
+        self, name: str, value: str, *, part_name: Optional[str] = None
+    ) -> None:
+        """Set the value of a project variable.
 
         :param name: The custom argument name.
         :param value: The new custom argument value.
 
         :raise ValueError: If there is no custom argument with the given name.
         """
-        if name not in self._custom_args:
-            raise ValueError(f"{name!r} not in project custom arguments")
+        self._ensure_valid_variable_name(name)
+
+        if name in self._consumed_project_vars:
+            raise ValueError(
+                f"cannot set variable {name!r}, it has already been consumed"
+            )
+
+        if self._project_vars_part_name in [None, part_name]:
+            self._project_vars[name] = value
+
+    def get_project_var(self, name: str, *, consume: bool = True) -> str:
```

Comment:
this leaked

Reply:
The method is correct, however the docstring was copy-and-pasted from a different method (it says "set" instead of "get")

Reply:
This method is intended to be used by the application to consume project variables. Setting variables after they're consumed must result in error to prevent unexpected/confusing situations when a variable is set after being consumed and the new value is not actually used.